### PR TITLE
bazel: Update `CODEOWNERS`, move `bazel` into `misc`

### PR DESCRIPTION
**Pull request by ParkMyCar**
This PR updates the `CODEOWNERS` file so folks will automatically get assigned to review Bazel related PRs.

@guswynn and @maddyblue I added y'all since you've been reviewing PRs thus far. If you don't want to be marked as an owner that's totally fine! Will wait for both of you to approve before merging

### Motivation

Automatic reviewers for PRs

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered. ([trigger-ci for additional test/nightly runs](https://trigger-ci.dev.materialize.com/))
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [x] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  - N/a

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -11,6 +11,7 @@
 /.cargo                             @benesch
 /.config                            @benesch
 /.github                            @benesch
+/bazel                              @parkmycar @guswynn @maddyblue
```
Comment (benesch) [inline .github/CODEOWNERS:14]: Sorry, didn't see the original PRs go past, but if Bazel doesn't care, can we put this somewhere besides the root of the repo? This is what `misc` was made for!

Comment (ParkMyCar) [reply .github/CODEOWNERS]: Totally, updated!

**Review by benesch — APPROVED** [review]
Thank you!

**Review by guswynn — APPROVED** [review]
perfetto!

**Review by maddyblue — APPROVED** [review]

**Pull request merged**